Cmake configuration failed for cross-compilation

Cmake configuration failed when I tried cross-compilation, because the execution `find_package(OpenCL REQUIRED)` failed. 
In this execution, cmake tries to compile some c program using `CMAKE_C_COMPILER`.
The problem is, this compiler run without any compiler-option to add include directories.
So, even though these exists `${CROSS_COMPILE}/include/CL`, finding OpenCL fails.

Probably `CMakeLists.txt` need to be modified.

My configuration command is as follows. Do I miss some specification?
Of course, `/opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/include/CL` exists.
`/opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -I/opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/include` works correctly.

```
cmake -DBUILD_TESTING=ON -DCROSS_COMPILE=ON -DCROSS_COMPILER_PATH=/opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf
```

My mistake: not in `/opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/include/CL`, but `/opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/arm-linux-gnueabihf/include/CL`

 ```
$ :~/VC4CL$ rm -rf CMakeCache.txt  CMakeFiles; cmake -DBUILD_TESTING=ON -DCROSS_COMPILE=ON -DCROSS_COMPILER_PATH=/opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Cross compiling for Raspbian with compiler: /opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf
-- The C compiler identification is GNU 7.1.1
-- The CXX compiler identification is GNU 7.1.1
-- Check for working C compiler: /opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc
-- Check for working C compiler: /opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-g++
-- Check for working CXX compiler: /opt/gcc-linaro-7.1.1-2017.08-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for CL_VERSION_2_0
-- Looking for CL_VERSION_2_0 - found
CMake Error at /usr/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenCL (missing: OpenCL_LIBRARY) (found version "2.0")
Call Stack (most recent call first):
  /usr/share/cmake-3.9/Modules/FindPackageHandleStandardArgs.cmake:377 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.9/Modules/FindOpenCL.cmake:132 (find_package_handle_standard_args)
  CMakeLists.txt:58 (find_package)

-- Configuring incomplete, errors occurred!
See also "/home/nomaddo/work/VC4CL/CMakeFiles/CMakeOutput.log".
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,14 +18,14 @@ option(CROSS_COMPILE "Cross compile for Raspbian" OFF)
 
 if(CROSS_COMPILE)
 	if(NOT CROSS_COMPILER_PATH)
-		set(CROSS_COMPILER_PATH "/opt/rasperrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64")
+		set(CROSS_COMPILER_PATH "/opt/rasberrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64")
 	endif()
 	message(STATUS "Cross compiling for Raspbian with compiler: ${CROSS_COMPILER_PATH}")
 	#Set Cross compiler
 	SET(CMAKE_SYSTEM_NAME 		"Linux")
 	SET(CMAKE_C_COMPILER   		"${CROSS_COMPILER_PATH}/bin/arm-linux-gnueabihf-gcc")
 	SET(CMAKE_CXX_COMPILER 		"${CROSS_COMPILER_PATH}/bin/arm-linux-gnueabihf-g++")
-	SET(CMAKE_FIND_ROOT_PATH  	"${CROSS_COMPILER_PATH} ${CROSS_COMPILER_PATH}/arm-linux-gnueabihf")
+	SET(CMAKE_FIND_ROOT_PATH  	"${CROSS_COMPILER_PATH}/arm-linux-gnueabihf")
 endif()
 
 project (VC4CL VERSION 0.4)
@@ -129,6 +129,11 @@ if (BUILD_TESTING)
 			STEP_TARGETS 		build
 	  		EXCLUDE_FROM_ALL	TRUE
 	  		TIMEOUT 			30		#Timeout for downloads, in seconds
+  		        CMAKE_ARGS
+    		          -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    		          -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
+  		          -DCMAKE_FIND_ROOT_PATH=${CMAKE_FIND_ROOT_PATH}
+
 		)
 	endif()
 	add_subdirectory(test build/test)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,7 @@ option(CROSS_COMPILE "Cross compile for Raspbian" OFF)
 
 if(CROSS_COMPILE)
 	if(NOT CROSS_COMPILER_PATH)
-		set(CROSS_COMPILER_PATH "/opt/rasberrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64")
+		set(CROSS_COMPILER_PATH "/opt/raspberrypi/tools/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian-x64")
 	endif()
 	message(STATUS "Cross compiling for Raspbian with compiler: ${CROSS_COMPILER_PATH}")
 	#Set Cross compiler

